### PR TITLE
C42f refactor map + various improvements

### PR DESCRIPTION
**Pull request by SimonDanisch**

### CODE REVIEW

**Comment by SimonDanisch** [comment]
@c42f thanks ;)

**Comment by SimonDanisch** [comment]
I seem to have angered the Github by hijacking your PR... Not sure what would have been the right process here ;)

**Comment by c42f** [comment]
I'm not sure github is angered - we simply have two separate PRs here containing partially the same code ;-)

AFAIK the only simple collaboration process for PRs is to both have access to the source branch - ie commit access to the same repository.  Other than that, we can pull changes into each other's branches which is quite simple.

I like the changes you're adding - they all seem like great cleanup work.  Personally I'd have kept some of the reformatting work in a separate PR (eg, the `symbol->Symbol` changes, whitespace changes).  It's totally worthwhile, but tends to create confusing clutter when reading the diffs to check on changes to actual functionality.

How much more of the work do you want to do for this?  What you've done here will definitely help a lot and I can continue with it by pulling more of your changes into my branch in #105.  It doesn't bother me much if I'm not the one to finish it off, but I'll keep plodding along with the promotion fixes if you're done with it for the time being.

**Comment by SimonDanisch** [comment]
Okay let me clean this up and merge it!

**Comment by SimonDanisch** [comment]
Also, I added you and @mschauer as collaborators to make things easier ;)
If you want to implement a more thorough PR process, feel free to bring it up!

**Comment by c42f** [comment]
Oh sure, thanks!  I'll probably keep committing to my branch while I experiment to avoid conflicts, but feel free to pull things in here or whatever.  I'm not sure how far I'll get anyway.

File: src/FixedSizeArrays.jl [hunk 1]
```diff
@@ -2,7 +2,10 @@ VERSION >= v"0.4.0-dev+6521" && __precompile__(true)
 module FixedSizeArrays
 
 importall Base
-import Base.Func, Base.LinAlg.chol!
+import Base.LinAlg.chol!
+
+# for 0.5 and 0.4 compat, use our own functor type
+abstract Functor{N}
```
Comment (c42f) [inline src/FixedSizeArrays.jl:8]: I was thinking about doing this as part of the promotion changes to ops.jl, but now I'm wondering if it'll interfere with our ability to make `a + b` return the right type when `a::Vector{Point{N,T}}` and `b::Vector{Vec{N,T}}`?

**Pull request closed without merging**